File: Programmers/[17676] 추석 트래픽/JFe/17676.py
from decimal import Decimal
def solution(lines):
    answer, log = 0, []
    for l in lines:
        l = l.split(' ')    # 로그 split
        time = l[1].split(':')  # 로그 시간 부분 split
        last_time = Decimal(time[0])*3600 + Decimal(time[1])*60 + Decimal(time[2])  # 응답 완료 시간 초 단위로 변환
        start_time = last_time - Decimal(l[2][:-1]) + Decimal('0.001')  # 시작 시간 = 응답 완료 시간 - 처리 시간
        log.append((start_time, last_time))
    for i in range(len(log)):
        count = 0
        t = log[i][1]   # 응답 완료 시간
        for j in range(i, len(log)):
            # i 인덱스 뒤 로그들 시작 시간이 (i 로그의 응답 완료 시간 + 0.999s) 이하면 count 추가
            if log[j][0] <= t + Decimal('0.999'):
                count += 1
        answer = max(answer, count) # 최댓값 저장
    return answer

File: Programmers/[17676] 추석 트래픽/ONE/17676.java
import java.util.ArrayList;

class Log {
    int requestTime;
    int responseTime;

    public Log(int responseTime, int sec) {
        this.requestTime = responseTime - sec + 1;
        this.responseTime = responseTime;
    }
}

class Solution {
    public int solution(String[] lines){
        int answer = 0;
        ArrayList<Integer> timeList = new ArrayList<>();
        ArrayList<Log> logList = new ArrayList<>();

        for (String line : lines) {
            String[] tokens = line.split(" ");
            String[] timeLine = tokens[1].split(":");

            int hour = Integer.parseInt(timeLine[0]) * 1000 * 60 * 60;
            int minute = Integer.parseInt(timeLine[1]) * 1000 * 60;
            int second = (int) (Double.parseDouble(timeLine[2]) * 1000);
            int processingTime = (int) (Double.parseDouble(tokens[2].substring(0, tokens[2].length() - 1)) * 1000);

            Log log = new Log(hour + minute + second, processingTime);
            logList.add(log);
            timeList.add(log.requestTime);
            timeList.add(log.responseTime);
        }

        for (int time : timeList) {
            int left = time;
            int right = time + 999;
            int count = 0;

            for (Log log : logList) {
                if ((log.requestTime >= left && log.requestTime <= right)
                        || (log.responseTime >= left && log.responseTime <= right)
                        || (log.requestTime <= left && log.responseTime >= right))
                    count++;
            }
            answer = Math.max(answer, count);
        }

        return answer;
    }
}